angular-ngrx-scss/src/app/auth/services/auth.service.spec.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { environment } from 'src/environments/environment';
+import { SignInResponse, AuthResponse } from '../interfaces/auth';
+import { AuthService } from './auth.service';
+import { TokenService } from './token.service';
+
+describe('AuthService', () => {
+  let tokenService: jasmine.SpyObj<TokenService>;
+  let authService: AuthService;
+  let httpController: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
+    tokenService = jasmine.createSpyObj('TokenService', ['getToken']);
+    authService = TestBed.inject(AuthService);
+    httpController = TestBed.inject(HttpTestingController);
+  });
+
+  it('should redirect the user to the sign in link', () => {
+    const expectedSignInResponse = {
+      redirectUrl: `${environment.apiUrl}/auth/signin`,
+    };
+
+    authService.signIn().subscribe((res) => {
+      expect(res).toEqual(expectedSignInResponse as SignInResponse);
+    });
+
+    const req = httpController.expectOne({
+      method: 'GET',
+      url: expectedSignInResponse.redirectUrl,
+    });
+
+    req.flush(expectedSignInResponse);
+  });
+
+  it('should get an access token', () => {
+    const expectedAuthResponse = {
+      access_token: '12345',
+      bearer: '',
+      scope: '',
+    };
+    const expectedPostUrl = `${environment.apiUrl}/auth/signin/callback`;
+
+    authService.getToken('code').subscribe((res) => {
+      expect(res).toEqual(expectedAuthResponse as AuthResponse);
+    });
+
+    const req = httpController.expectOne({
+      method: 'POST',
+      url: expectedPostUrl,
+    });
+
+    req.flush(expectedAuthResponse);
+  });
+
+  it('checks if a user is authenticated', () => {
```

Comment:
@TapaiBalazs Would love your help here - the test suite doesn't always make these mock calls in order or run these in order, so the test sometimes passes and sometimes doesn't. Is there a better way I can more consistently give it the value I want to test the functionality?